crates/ruff_linter/src/rules/pyupgrade/rules/convert_typed_dict_functional_to_class.rs

```diff
@@ -185,7 +184,7 @@ fn fields_from_dict_literal(items: &[ast::DictItem]) -> Option<Vec<Stmt>> {
                     if !is_identifier(field.to_str()) {
                         return None;
                     }
-                    if is_dunder(field.to_str()) {
+                    if field.to_str().starts_with("__") {
```

Comment:
I think it would be good to add a comment here explaining why it's not safe to convert TypedDicts with these fields to the class-based syntax. And it might be useful to add a sentence or two to the docs explaining why not all functional TypedDicts can be converted (and that therefore this rule deliberately does not flag them), as well!

Reply:
Done